From: Mike Hommey <mh@glandium.org>
Date: Mon, 16 Dec 2013 10:53:58 +0900
Subject: Add debian/extra-stuff to upstream build system directory traversal

---
 moz.build | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/moz.build b/moz.build
index 61f6594c8ec..92807fdf45f 100644
--- a/moz.build
+++ b/moz.build
@@ -200,4 +200,7 @@ SPHINX_TREES['bug-mgmt'] = 'docs/bug-mgmt'
 
 SPHINX_TREES['setup'] = 'docs/setup'
 
+if not CONFIG['LIBXUL_SDK']:
+    DIRS += ['debian/extra-stuff']
+
 include('build/templates.mozbuild')
